### PR TITLE
Added support for `warehouse_id`, `options`, `partitions` and liquid clustering for `databricks_sql_table` resource

### LINKED ISSUE #2372: [FEATURE] Support SQL warehouses for CRUD operations on databricks_sql_table
### Use-cases

CRUD operations to a `databricks_sql_table` require a Compute cluster (the `cluster_id` argument). The majority of the time I feel SQL operations like this are done from SQL Warehouses, not all purpose clusters. As an added benefit, with the addition of SQL Serverless, this would greatly speed up execution for creating tables from a "cold start".

This feature request is to be able to use the `databricks_sql_table` resource with SQL Warehouses instead of Compute Clusters. 

### Attempted Solutions

I initially attempted to use a SQL warehouse ID in the `cluster_id` argument for the `databricks_sql_table` resource:

```terraform
resource "databricks_sql_table" "this" {
  catalog_name       = databricks_schema.this.catalog_name
  schema_name        = databricks_schema.this.name
  name               = "mytable"
  data_source_format = "PARQUET"
  table_type         = "EXTERNAL"
  storage_location   = var.storage_location
  cluster_id         = var.sql_warehouse_id # only works with Compute Cluster IDs, not SQL warehouses
  properties = {
    "tags.DeployedBy" = "Terraform"
  }
}
```

Provider version: 1.18.0
```
provider[registry.terraform.io/databricks/databricks] >= 1.18.0
```

### Proposal

Because there could be ambiguity differentiating Compute Cluster IDs from SQL Warehouse IDs, I'd like to propose a new optional string argument `warehouse_id` which should be mutually exclusive from `cluster_id`. And therefore, also should have backwards compatibility with the existing provider. 

For example, the usage above would become:

```terraform
resource "databricks_sql_table" "this" {
  catalog_name       = databricks_schema.this.catalog_name
  schema_name        = databricks_schema.this.name
  name               = "mytable"
  data_source_format = "PARQUET"
  table_type         = "EXTERNAL"
  storage_location   = var.storage_location
  # cluster_id = "" <-- cannot use this argument if using the `warehouse_id` argument
  warehouse_id         = var.sql_warehouse_id
  properties = {
    "tags.DeployedBy" = "Terraform"
  }
}
```

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes
- Add support for `warehouse_id`, `options`, `partitions` and liquid clustering for `databricks_sql_table` resource

Closes #2372, #2639, #2721

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

File: catalog/resource_sql_table.go [hunk 1]
```diff
@@ -75,6 +82,17 @@ func sqlTableIsManagedProperty(key string) bool {
 		"delta.lastUpdateVersion":                                  true,
 		"delta.minReaderVersion":                                   true,
 		"delta.minWriterVersion":                                   true,
+		"delta.enableDeletionVectors":                              true,
```
Comment (tanmay-db) [inline catalog/resource_sql_table.go:85]: Curious -- from where are we getting these map keys?

**Review by tanmay-db — APPROVED** [review]

File: catalog/resource_sql_table.go [hunk 2]
```diff
@@ -302,12 +352,29 @@ func (ti *SqlTableInfo) deleteTable() error {
...
 	log.Printf("[INFO] Executing Sql: %s", sqlQuery)
-	r := ti.exec.Execute(ti.ClusterID, "sql", sqlQuery)
-
-	if !r.Failed() {
+	if ti.WarehouseID != "" {
+		execCtx, cancel := context.WithTimeout(context.Background(), 50*time.Second)
+		defer cancel()
+		sqlRes, err := ti.sqlExec.ExecuteStatement(execCtx, sql.ExecuteStatementRequest{
+			Statement:     sqlQuery,
+			WaitTimeout:   "50s", //max allowed by sql exec
```
[the remark this answers is not in the document]
Comment (nkvuong) [reply catalog/resource_sql_table.go]: refactored

File: catalog/resource_sql_table.go [hunk 3]
```diff
@@ -29,14 +33,19 @@ type SqlTableInfo struct {
 	TableType             string            `json:"table_type" tf:"force_new"`
 	DataSourceFormat      string            `json:"data_source_format,omitempty" tf:"force_new"`
 	ColumnInfos           []SqlColumnInfo   `json:"columns,omitempty" tf:"alias:column,computed,force_new"`
+	Partitions            []string          `json:"partitions,omitempty" tf:"force_new"`
+	ClusterKeys           []string          `json:"cluster_keys,omitempty" tf:"force_new"`
```
Comment (alexott) [inline catalog/resource_sql_table.go:37]: I'm not sure about the `force_new` - you can change cluster keys without dropping a table.

File: catalog/resource_sql_table.go (hunk 1 above)
Comment (tanmay-db) [reply catalog/resource_sql_table.go:87]: I see

File: catalog/resource_sql_table.go [hunk 4]
```diff
@@ -29,14 +33,19 @@ type SqlTableInfo struct {
...
 	StorageLocation       string            `json:"storage_location,omitempty" tf:"suppress_diff"`
 	StorageCredentialName string            `json:"storage_credential_name,omitempty" tf:"force_new"`
 	ViewDefinition        string            `json:"view_definition,omitempty"`
 	Comment               string            `json:"comment,omitempty"`
 	Properties            map[string]string `json:"properties,omitempty" tf:"computed"`
+	Options               map[string]string `json:"options,omitempty" tf:"force_new"`
 	ClusterID             string            `json:"cluster_id,omitempty" tf:"computed"`
+	WarehouseID           string            `json:"warehouse_id,omitempty"`
```
Comment (alexott) [inline catalog/resource_sql_table.go:45]: Should cluster ID conflict with warehouse ID?

**Pull request merged**